### ISSUE TITLE
Make text object coordinates more bulletproof ?

### ISSUE THREAD

**Issue by jypeter** [UV-CDAT/uvcdat#1078]
I had problems with a script where I could plot a text object in 1.5.1, but I got the following error in 2.1.0

```
Traceback (most recent call last):
  File "test_text_method.py", line 27, in <module>
    x.plot(text_test)
  File "/home/share/unix_files/cdat/versions/cdat_install_uv-2.1.0_x86_64_gcc4_13/lib/python2.7/site-packages/vcs/Canvas.py", line 2589, in plot
    a = self.__plot( arglist, keyargs )
  File "/home/share/unix_files/cdat/versions/cdat_install_uv-2.1.0_x86_64_gcc4_13/lib/python2.7/site-packages/vcs/Canvas.py", line 3649, in __plot
    returned_kargs = self.backend.plot(*arglist,**keyargs)
  File "/home/share/unix_files/cdat/versions/cdat_install_uv-2.1.0_x86_64_gcc4_13/lib/python2.7/site-packages/vcs/VTKPlots.py", line 410, in plot
    vcs2vtk.genTextActor(ren,to=to,tt=tt)
  File "/home/share/unix_files/cdat/versions/cdat_install_uv-2.1.0_x86_64_gcc4_13/lib/python2.7/site-packages/vcs/vcs2vtk.py", line 872, in genTextActor
    X,Y = world2Renderer(renderer,x[i],y[i],tt.viewport,tt.worldcoordinate)
  File "/home/share/unix_files/cdat/versions/cdat_install_uv-2.1.0_x86_64_gcc4_13/lib/python2.7/site-packages/vcs/vcs2vtk.py", line 1226, in world2Renderer
    X = origin[0]+ (opposite[0]-origin[0] )*(x-wc[0])/(wc[1]-wc[0])
TypeError: unsupported operand type(s) for -: 'list' and 'int'
```

It turned out that **1.5.1 is happy with coordinates specified in either of the following ways**
 1. text_test.x = [20., 40.]
 2. text_test.x = [[20.], [40.]]

But **2.1.0 can only deal with 1)** and will throw the exception above with 2)

Can you update the function to make it more bulletproof, so that 2.1.0 can also handle format 2)

The following test script can be used to reproduce the error

```
#!/usr/bin/env python

import vcs

#coord_cdat210_OK = True
coord_cdat210_OK = False

x = vcs.init()

x.text('Plotting directly to the canvas',.2,.5, color=242, angle=0)

text_test = x.createtext('marker_names')
text_test.viewport = [0.05, 0.95, 0.1, 0.9]
text_test.worldcoordinate = [0, 100, 0, 100]
if coord_cdat210_OK:
    # Specifying text coordinates like this works for 1.5.1 AND for 2.1.0
    text_test.x = [20., 40.]
    text_test.y = [30., 20.]
else:
    # Specifying text coordinates like this works for 1.5.1 but NOT for 2.1.0
    text_test.x = [[20.], [40.]]
    text_test.y = [[30.], [20.]]
text_test.string = ['1st string of text_test',
                    '2nd string of text_test']
text_test.color = 248

x.plot(text_test)

# The end
```

**Comment by dlonie** [comment]
Seems odd that we would want to support arbitrary nesting of lists. It complicates things considerably on the backend, as every access into this array has to support all of the different formats.

Is there a usecase where coordinates being specified in this way would be more useful than the equivalent flat list?

**Comment by jypeter** [comment]
It's odd to get nested list, but sometimes you need nested lists for some vcs objects properties and sometimes you don't, which is confusing. That's for instance the case in the test script I provided for #1079 where you have **groups of several points that have the same marker type/size/color**. I agree it is very confusing, and only documented in the vcs objects (or x.createXXXXX) docstrings. @chaosphere2112 , that's one more thing that should be documented on the web site... :-/
```
marker_test.x = [[90, 90, 90],
                 [0, 10, 20, 30, 40, 50, 60, 70, 80, 90, 100],
                 [0, 10, 20, 30, 40, 50, 60, 70, 80, 90, 100]]
marker_test.y = [[25, 50, 75],
                 [0, 10, 20, 30, 40, 50, 60, 70, 80, 90, 100],
                 [100, 90, 80, 70, 60, 50, 40, 30, 20, 10, 0]]
marker_test.type = ['triangle_down_fill',
                    'diamond_fill',
                    'dot']
marker_test.size = [20.,
                    20.,
                    15.]
marker_test.color = [243, # Green
                     248, # Orange
                     241] # Black
````

**Comment by chaosphere2112** [comment]
Yup, it's on my list of things to add to the documentation (a thorough primer on each of the secondary objects, and how each of their attributes work). I've written up fill areas so far, haven't had time for a documentation sprint in a bit. I'm planning to have some time to fix up web/admin/doc issues in the next week or two.

**Comment by doutriaux1** [comment]
I think 2.1 actually fixes a bug in 1.5 only marker are truly designed to manage nested list (and I don't like this).
